CondTools/Ecal/src/EcalTimeCalibHandler.cc

```diff
@@ -0,0 +1,123 @@
+#include "CondTools/Ecal/interface/EcalTimeCalibHandler.h"
+#include "FWCore/ParameterSet/interface/ParameterSetfwd.h"
+#include "CondFormats/EcalObjects/interface/EcalTimeCalibConstants.h"
+#include "CondTools/Ecal/interface/EcalFloatCondObjectContainerXMLTranslator.h"
+
+#include <iostream>
+
+const Int_t kEBChannels = 61200, kEEChannels = 14648;
+
+popcon::EcalTimeCalibHandler::EcalTimeCalibHandler(const edm::ParameterSet& ps)
+    : m_name(ps.getUntrackedParameter<std::string>("name", "EcalTimeCalibHandler")) {
+  edm::LogInfo("EcalTimeCalib Source handler constructor\n");
+  m_firstRun = static_cast<unsigned int>(atoi(ps.getParameter<std::string>("firstRun").c_str()));
+  m_file_type = ps.getParameter<std::string>("type");  // xml/txt
+  m_file_name = ps.getParameter<std::string>("fileName");
+}
+
+popcon::EcalTimeCalibHandler::~EcalTimeCalibHandler() {}
+
+void popcon::EcalTimeCalibHandler::getNewObjects() {
+  //  std::cout << "------- Ecal - > getNewObjects\n";
+  std::ostringstream ss;
+  ss << "ECAL ";
+
+  unsigned long long irun;
+  std::string file_ = m_file_name;
+  edm::LogInfo("going to open file ") << file_;
+
+  //      EcalCondHeader   header;
+  EcalTimeCalibConstants* payload = new EcalTimeCalibConstants;
+  if (m_file_type == "xml")
+    readXML(file_, *payload);
+  else
+    readTXT(file_, *payload);
+  irun = m_firstRun;
+  Time_t snc = (Time_t)irun;
```

Comment:
Is the intermediate `irun` needed? IOW, why not more simply
```suggestion
  Time_t snc = (Time_t)m_firstRun;
```